### PR TITLE
KOGITO-7575 - Add persistence related config properties to Quarkus

### PR DESCRIPTION
**Pull request by cristianonicolai**
Fixes https://github.com/kiegroup/kogito-runtimes/issues/2324

### CODE REVIEW

**Comment by kie-ci3** [comment]
**(tests) - kogito-runtimes job** `#471` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/471/display/redirect

**Test results:**
- PASSED: 2819
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/471/testReport/org.kie.kogito.it/MongoDBOptimisticLockingIT/Build_projects___testPersistence/">Build projects / org.kie.kogito.it.MongoDBOptimisticLockingIT.testPersistence</a></summary>
java.lang.RuntimeException: java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details<br/>Caused by: java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details<br/>
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/471/testReport/org.kie.kogito.it/MongoDBPersistenceIT/Build_projects___testPersistence/">Build projects / org.kie.kogito.it.MongoDBPersistenceIT.testPersistence</a></summary>
java.lang.RuntimeException: java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details<br/>Caused by: java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details<br/>
</details>

**Comment by kie-ci3** [comment]
**(tests) - kogito-examples job** `#517` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/517/display/redirect

**Test results:**
- PASSED: 521
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/517/testReport/org.acme.deals/DealsRestIT/Build_projects___testDealsRest/">Build projects / org.acme.deals.DealsRestIT.testDealsRest</a></summary>
java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus<br/>Caused by: java.lang.RuntimeException: Failed to start quarkus<br/>Caused by: io.quarkus.runtime.configuration.ConfigurationException: <br/>One or more configuration errors have prevented the application from starting. The errors are:<br/>  - SRCFG00039: The config property kogito.persistence.type with the config value "mongodb" threw an Exception whilst being converted Cannot convert mongodb to enum class org.kie.kogito.quarkus.conf.KogitoPersistenceRuntimeConfig$PersistenceType<br/><br/>
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/517/testReport/org.kie.kogito.quarkus.outbox/OutboxIT/___/">org.kie.kogito.quarkus.outbox.OutboxIT.(?)</a></summary>
java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [8080, 49183] should be listening)<br/>	at org.kie.kogito.quarkus.outbox.OutboxIT.before(OutboxIT.java:86)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [8080, 49183] should be listening)<br/>
</details>

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#604` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-apps/604/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[2022-07-15T07:04:26.622Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-core/node_modules/@patternfly/react-tokens/dist/js/chart_axis_grid_PointerEvents.d.ts’: No such file or directory<br/>[2022-07-15T07:04:26.622Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-core/node_modules/@patternfly/react-tokens/dist/js/chart_axis_grid_Fill.js’: No such file or directory<br/>[2022-07-15T07:04:26.622Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-core/node_modules/@patternfly/react-tokens/dist/js/chart_axis_grid_PointerEvents.js’: No such file or directory<br/>[2022-07-15T07:04:26.622Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-core/node_modules/@patternfly/react-tokens/dist/js/chart_axis_grid_stroke_Color.d.ts’: No such file or directory<br/>[2022-07-15T07:04:26.622Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-core/node_modules/@patternfly/react-tokens/dist/js/chart_axis_grid_stroke_Color.js’: No such file or directory<br/>[2022-07-15T07:04:26.622Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-core/node_modules/@patternfly/react-tokens/dist/js/chart_axis_tick_Size.d.ts’: No such file or directory<br/>[2022-07-15T07:04:26.877Z] 2.2G	/home/jenkins/workspace<br/>[2022-07-15T07:04:26.877Z] 0	/home/jenkins/.pki<br/>[2022-07-15T07:04:26.877Z] 137M	/home/jenkins/tools<br/>[2022-07-15T07:04:26.877Z] 0	/home/jenkins/caches<br/>[2022-07-15T07:04:26.877Z] 28M	/home/jenkins/.npm<br/>[2022-07-15T07:04:26.877Z] 629M	/home/jenkins/.gradle<br/>[2022-07-15T07:04:26.877Z] 0	/home/jenkins/.yarn<br/>[2022-07-15T07:04:26.877Z] 7.2G	/home/jenkins<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Install build-chain tool)<br/>Stage "Install build-chain tool" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-07-15T07:04:27.797Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-apps/604/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by kie-ci3** [comment]
**(tests) - kogito-examples job** `#520` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/520/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[2022-07-15T07:04:35.666Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/node_modules/rxjs/observable/concat.js.map’: No such file or directory<br/>[2022-07-15T07:04:35.666Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/node_modules/rxjs/observable/defer.d.ts’: No such file or directory<br/>[2022-07-15T07:04:35.666Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/node_modules/rxjs/observable/defer.js’: No such file or directory<br/>[2022-07-15T07:04:35.666Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/node_modules/rxjs/observable/defer.js.map’: No such file or directory<br/>[2022-07-15T07:04:35.666Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/node_modules/rxjs/observable/empty.d.ts’: No such file or directory<br/>[2022-07-15T07:04:35.666Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/main/pullrequest.quarkus-lts/kogito-runtimes.quarkus-lts.downstream.kogito-apps_ws-cleanup_1657868648164/bc/kiegroup_kogito_apps/kogito-apps/ui-packages/node_modules/rxjs/observable/empty.js’: No such file or directory<br/>[2022-07-15T07:04:35.666Z] 187M	/home/jenkins/workspace<br/>[2022-07-15T07:04:35.666Z] 0	/home/jenkins/.pki<br/>[2022-07-15T07:04:35.666Z] 137M	/home/jenkins/tools<br/>[2022-07-15T07:04:35.666Z] 0	/home/jenkins/caches<br/>[2022-07-15T07:04:35.666Z] 28M	/home/jenkins/.npm<br/>[2022-07-15T07:04:35.666Z] 629M	/home/jenkins/.gradle<br/>[2022-07-15T07:04:35.666Z] 0	/home/jenkins/.yarn<br/>[2022-07-15T07:04:35.666Z] 5.2G	/home/jenkins<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Install build-chain tool)<br/>Stage "Install build-chain tool" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-07-15T07:04:36.479Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/520/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-apps tests

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-examples tests

File: quarkus/extensions/kogito-quarkus-extension-common/kogito-quarkus-common/src/main/java/org/kie/kogito/quarkus/conf/KogitoPersistenceRuntimeConfig.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+     */
+    @ConfigItem
+    public Optional<PersistenceType> type;
+
+    /**
+     * Automatically apply database schema changes
+     */
+    @ConfigItem(name = "auto.ddl", defaultValue = "true")
+    public boolean autoDDL;
+
```
Comment (nmirasch) [inline quarkus/extensions/kogito-quarkus-extension-common/kogito-quarkus-common/src/main/java/org/kie/kogito/quarkus/conf/KogitoPersistenceRuntimeConfig.java:38]: @cristianonicolai I miss  this one here:   @ConfigItem(name = "query.timeout.millis", defaultValue = "10000")
Some examples are complaining about this one :-)

Comment (cristianonicolai) [reply quarkus/extensions/kogito-quarkus-extension-common/kogito-quarkus-common/src/main/java/org/kie/kogito/quarkus/conf/KogitoPersistenceRuntimeConfig.java:38]: @nmirasch nice catch, thanks

**Review by fjtirado — APPROVED** [review]
One question, the properties defined under persistence, how are accesses later in the code, through KogitoBuildContext.getApplicationProperty?

**Comment by fjtirado** [comment]
> > dContext.getApplication
> 
> yep, we should migrate to use these config classes instead but would be another set of work

I need to test if, at runtime (and only at runtime), when using native, the variable being accessed needs to be defined as configroot. Maybe it is not and we must add to the configbean only those ones accessed during build time (by the extension).  Or do we want to add all known properties (regardless they are really needed there in order to work in native) regardless of the result of that test?

**Comment by cristianonicolai** [comment]
I think the Quarkus way to do this is to add all properties to the configroot and use that bean in the extension processor. That would mean not using `KogitoBuildContext.getApplicationProperty` in the extension but using the configroot object directly. It might need some tweaking to make that work with codegen modules.

**Pull request merged**